tests/ui/imports/issue-56125.stderr

```diff
@@ -8,13 +8,10 @@ help: consider importing one of these items instead
    |
 LL |     use ::issue_56125::issue_56125;
    |         ~~~~~~~~~~~~~~~~~~~~~~~~~~
-LL |     use ::issue_56125::last_segment::issue_56125;
-   |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-LL |     use ::issue_56125::non_last_segment::non_last_segment::issue_56125;
```

Comment:
I'm not sure whether we should remove some of them, they are removed by:

```rust
for item in suggest_imports {
                        if !suggestions.iter().any(|sugg| sugg.did == item.did) {
                            suggestions.push(item);
                        }
                    }
```

Seems it less noise after removing them.

Reply:
it seems this will make the CI fail, I can not reproduce it locally.
why there are different behavior between different arch, not quite understand it.

Reply:
This deduplication seems like a separate change to what the issue reported, and I presume that this is what removes the `std::` suggestions in favour of `core::` suggestions, which I don't think is an improvement.